goal_src/engine/nav/path.gc

```diff
@@ -5,3 +5,496 @@
+       )
+      (vector-! arg0 arg0 s4-0)
+      )
+     )
+    )
+   )
+  (vector-normalize! arg0 1.0)
+  )
+
+;; definition for method 12 of type curve-control
+(defmethod
+  TODO-RENAME-12
+  curve-control
+  ((obj curve-control) (arg0 vector) (arg1 float))
+  (TODO-RENAME-14
+   obj
+   arg0
+   (/ arg1 (the float (+ (-> obj curve num-cverts) -1)))
+   )
+  )
+
+;; definition for method 19 of type path-control
+;; Used lq/sq
+(defmethod TODO-RENAME-19 path-control ((obj path-control))
+  (let ((s5-0 (new 'stack-no-clear 'vector))
+        (s4-0 (new 'stack-no-clear 'vector))
+        (s3-0 (new 'stack-no-clear 'vector))
+        (f30-0 4096000000.0)
+        (f28-0 0.0)
+        )
+   (let ((s2-0 (new 'stack-no-clear 'vector)))
+    (set! (-> s3-0 quad) (-> (target-pos 0) quad))
+    (set! (-> s3-0 y) 0.0)
+    (eval-path-curve-div! obj s4-0 0.0 'interp)
+    (set! (-> s4-0 y) 0.0)
+    (dotimes (s1-1 (+ (-> obj curve num-cverts) -1))
+     (set! (-> s5-0 quad) (-> s4-0 quad))
+     (eval-path-curve-div! obj s4-0 (the float (+ s1-1 1)) 'interp)
+     (set! (-> s4-0 y) 0.0)
+     (let ((f0-5 (vector-segment-distance-point! s3-0 s5-0 s4-0 s2-0)))
+      (when (< f0-5 f30-0)
+       (set! f30-0 f0-5)
+       (set!
+        f28-0
+        (+
+         (/
+          (vector-vector-xz-distance s2-0 s5-0)
+          (vector-vector-xz-distance s4-0 s5-0)
+          )
+         (the float s1-1)
+         )
+        )
+       )
+      )
+     )
+    )
+   f28-0
+   )
+  )
+
+;; definition for method 20 of type path-control
+(defmethod TODO-RENAME-20 path-control ((obj path-control))
+  (/ (TODO-RENAME-19 obj) (the float (+ (-> obj curve num-cverts) -1)))
+  )
+
+;; definition for method 9 of type curve-control
+;; INFO: Return type mismatch int vs none.
+(defmethod dummy-9 curve-control ((obj curve-control))
+  (cond
+   ((nonzero? (logand (-> obj flags) (path-control-flag not-found)))
+    (when
+     (and
+      (type-type? (-> obj process type) process-drawable)
+      *display-entity-errors*
+      )
+     (let ((s5-0 add-debug-text-3d)
+           (s4-0 #t)
+           (s3-0 68)
+           )
+      (format
+       (clear *temp-string*)
+       "curve data error in ~S"
+       (-> obj process name)
+       )
+      (s5-0
+       s4-0
+       (the-as bucket-id s3-0)
+       *temp-string*
+       (-> obj process root trans)
+       (new 'static 'rgba :r #x3)
+       (the-as vector2h #f)
+       )
+      )
+     )
+    )
+   ((let ((a0-5 obj))
+     (and
+      *display-path-marks*
+      (nonzero? (logand (-> a0-5 flags) (path-control-flag display)))
+      )
+     )
+    (when
+     (and
+      (nonzero? (logand (-> obj flags) (path-control-flag draw-line)))
+      (> (-> obj curve num-cverts) 0)
+      )
+     (let ((t9-4 add-debug-curve2)
+           (a0-8 #t)
+           (a1-3 68)
+           (a2-2 (-> obj curve))
+           (a3-1 (new 'static 'rgba :r #xff :g #x80 :a #x80))
+           )
+      (t9-4 a0-8 (the-as bucket-id a1-3) a2-2 a3-1)
```

Comment:
I think we have the type of `add-debug-curve2` wrong.  The asm for this is:
```asm
    lw t9, add-debug-curve2(s7)
    daddiu a0, s7, #t
    addiu a1, r0, 68
    daddiu a2, gp, 12
    ld a3, L79(fp)
    or t0, s7, r0
    jalr ra, t9
    sll v0, ra, 0
 ```

The `or t0, s7, r0` is setting a 5th argument to `#f`. I think if you add a `symbol` argument to `add-debug-curve2`, this will decompile correctly.